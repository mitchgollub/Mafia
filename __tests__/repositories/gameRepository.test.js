import GameRepository from '../../repositories/gameRepository';
import MongoDb from '../../lib/mongodb';

jest.mock('../../lib/mongodb');
Math.random = jest.fn(() => 0.48);

describe('GameRepository', () => {
  test('Creates Game', async () => {
    const roles = [
      { role: 'Cop', roleName: 'Cops', startingValue: 1 },
      { role: 'Mafia', roleName: 'Mafia', startingValue: 2 },
      { role: 'Doctor', roleName: 'Doctors', startingValue: 1 },
      { role: 'Town Watch', roleName: 'Town Watch', startingValue: 1 },
      { role: 'Townsfolk', roleName: 'Townsfolk', startingValue: 3 },
    ];

    const actual = await GameRepository.createGame(roles);
    expect(actual).toMatchSnapshot();
  });

  test('Gets Game', async () => {
    const code = 'MMMM';
    MongoDb.findGameDocument.mockResolvedValue({
      code: 'MMMM',
      players: {
        current: [{ id: 1, role: 'Narrator', name: 'YOU', session: '' }],
        available: [
          { id: 2, role: 'Cop' },
          { id: 3, role: 'Mafia' },
          { id: 4, role: 'Mafia' },
          { id: 5, role: 'Doctor' },
          { id: 6, role: 'Town Watch' },
          { id: 7, role: 'Townsfolk' },
          { id: 8, role: 'Townsfolk' },
          { id: 9, role: 'Townsfolk' },
        ],
      },
    });

    const actual = await GameRepository.getGame(code);
    expect(actual).toMatchSnapshot();
  });
});
