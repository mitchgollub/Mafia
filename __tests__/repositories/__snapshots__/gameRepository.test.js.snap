// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameRepository Creates Game 1`] = `
Game {
  "code": "MMMM",
  "players": Object {
    "available": Array [
      AvailableRole {
        "id": 2,
        "role": "Cop",
      },
      AvailableRole {
        "id": 3,
        "role": "Mafia",
      },
      AvailableRole {
        "id": 4,
        "role": "Mafia",
      },
      AvailableRole {
        "id": 5,
        "role": "Doctor",
      },
      AvailableRole {
        "id": 6,
        "role": "Town Watch",
      },
      AvailableRole {
        "id": 7,
        "role": "Townsfolk",
      },
      AvailableRole {
        "id": 8,
        "role": "Townsfolk",
      },
      AvailableRole {
        "id": 9,
        "role": "Townsfolk",
      },
    ],
    "current": Array [
      Object {
        "id": 1,
        "name": "YOU",
        "role": "Narrator",
        "session": "",
      },
    ],
  },
}
`;

exports[`GameRepository Gets Game 1`] = `
Object {
  "code": "MMMM",
  "players": Object {
    "available": Array [
      Object {
        "id": 2,
        "role": "Cop",
      },
      Object {
        "id": 3,
        "role": "Mafia",
      },
      Object {
        "id": 4,
        "role": "Mafia",
      },
      Object {
        "id": 5,
        "role": "Doctor",
      },
      Object {
        "id": 6,
        "role": "Town Watch",
      },
      Object {
        "id": 7,
        "role": "Townsfolk",
      },
      Object {
        "id": 8,
        "role": "Townsfolk",
      },
      Object {
        "id": 9,
        "role": "Townsfolk",
      },
    ],
    "current": Array [
      Object {
        "id": 1,
        "name": "YOU",
        "role": "Narrator",
        "session": "",
      },
    ],
  },
}
`;
